### DISCUSSION TITLE
rest api - full config for device

### DISCUSSION THREAD

**Discussion by zdenoblazicek** [emsesp/EMS-ESP32#1702] [Q&A]
Hello

how to get the whole configuration for the thermostat/boiler

if I call http://ems-eps.local/api/thermostat
and then I go through all the values in a loop, so MQTT for homeassistant stops working and I have to restart the EMS-EPS gateway
(growing MQTT Errors)

is it possible to call something like http://ems-eps.local/api/thermostat/config where the information would be
type, values and etc.

{
"config":[
{
   "name": "datetime",
   "fullname": "date/time",
   "circuit": "",
   "type": "string",
   "readable": true,
   "writeable": true,
   "visible": true
},
{
   "name": "intoffset",
   "fullname": "internal temperature offset",
   "circuit": "",
   "type": "number",
   "min": -12,
   "max": 12,
   "uom": "°C",
   "readable": true,
   "writeable": true,
   "visible": true
}
......
]
}

I would like to create a dynamically changing program for homey for 2023

**Comment by MichaelDvP** [comment]
> if I call http://ems-eps.local/api/thermostat and then I go through all the values in a loop, so MQTT for homeassistant stops working and I have to restart the EMS-EPS gateway (growing MQTT Errors)

I know that Thomas uses this methode for the ioBroker adapter, but without issue. Seems you are flooding the api with requests.
> is it possible to call something like http://ems-eps.local/api/thermostat/config where the information would be
type, values and etc.

Possible? Yes. But i think the json will get too large and we ran into memory issues on esp without psram. Low memory results in the same issue: mqtt errors.
